src/elements/ngm-nav-tools.ts

```diff
@@ -67,12 +94,95 @@ export class NgmNavTools extends LitElementI18n {
   }
 
   flyToHome() {
-    if (!this.scene) return;
-    this.scene.camera.flyTo(DEFAULT_VIEW);
+    if (!this.viewer) return;
+    this.viewer.camera.flyTo(DEFAULT_VIEW);
+  }
+
+  toggleReference() {
+    if (!this.eventHandler) return;
+    if (this.showRefPoint) {
+      this.eventHandler.removeInputAction(ScreenSpaceEventType.MOUSE_MOVE);
+      this.eventHandler.removeInputAction(ScreenSpaceEventType.LEFT_DOWN);
+      this.eventHandler.removeInputAction(ScreenSpaceEventType.LEFT_UP);
+      this.removeReference();
+    } else {
+      this.eventHandler.setInputAction(debounce(event => this.onMoveRef(event), 250), ScreenSpaceEventType.MOUSE_MOVE);
+      this.eventHandler.setInputAction(event => this.onLeftDown(event), ScreenSpaceEventType.LEFT_DOWN);
+      this.eventHandler.setInputAction(() => this.onLeftUp(), ScreenSpaceEventType.LEFT_UP);
+      this.setReference();
+    }
+  }
+
+  setReference(position?: Cartesian3) {
+    const center = position || pickCenterOnMapOrObject(this.viewer!.scene);
+    if (!center) {
+      showWarning(i18next.t('nav_tools_out_glob_warn'));
+      return;
+    }
+    this.showRefPoint = true;
+    this.refIcon.position = <any>center;
+    const cam = this.viewer!.camera;
+    this.refIcon.show = true;
+    this.refIcon.viewFrom = <any>eastNorthUp(center, cam.position);
+    if (!position) {
+      const transform = Transforms.eastNorthUpToFixedFrame(center);
+      cam.lookAtTransform(transform);
+    }
+    document.addEventListener('keydown', this.ctrlListener);
+
+  }
+
+  removeReference() {
```

Comment:
What about using the same wording as the user interface? Here and in the other places.
```suggestion
  removeTargetPoint() {
```